script/testing/util/test_case.py

```diff
@@ -0,0 +1,35 @@
+from util import constants
+from util.common import *
+
+class TestCase:
+    """Class of a test case, could be part of a test suite"""
+
+    def __init__(self, args):
+        """ Locations and misc. variable initialization """
+        # clean up the command line args
+        self.args = {k: v for k, v in args.items() if v}
+
+        # db server location
+        self.db_host = self.args.get("db_host", constants.DEFAULT_DB_HOST)
+        self.db_port = self.args.get("db_port", constants.DEFAULT_DB_PORT)
+
+        # test execution output
+        self.test_output_file = self.args.get(
+            "test_output_file", constants.DEFAULT_TEST_OUTPUT_FILE)
+
+        # test execution command
+        self.test_command = ""
```

Comment:
I think the test command can be `""` or `None` by default but we do need to check the string before execution. In case there is an empty string or `None` we then issue a warning (or an error if it is the only test case)